### PR TITLE
Switched from Cake to Nuke

### LINKED ISSUE #41: Build script fails and tests don't run in VS
Here's my output:

```ps1
**********************************************************************
** Visual Studio 2019 Developer PowerShell v16.6.5
** Copyright (c) 2020 Microsoft Corporation
**********************************************************************
Shimmy@SHIMMY-PC  D:\Users\Shimmy\Source\Repos\weitzhandler\fluentassertions.json   have-element-and-value ≣[09:31]
❯ ./build
Preparing to run build script...
Running build script...
Analyzing build script...
Processing build script...
Installing tools...
Downloading and installing Roslyn...
Installing packages (using https://packages.nuget.org/api/v2)...
Error: The underlying connection was closed: An unexpected error occurred on a send.
```

In addition, VS discoveres the the tests, but does not run any of them. After pressing <kbd>Ctrl</kbd>+<kbd>R</kbd>, <kbd>A</kbd>, it says:

> Outcomes: 59 Not Run.

### PR DESCRIPTION
**Pull request by dennisdoomen**
The master branch did not even build anymore, so took the opportunity to switch to Nuke. Also updated the unit test project to the new format and enabled multi-targeting.

Fixes #41

### CODE REVIEW

File: Tests/FluentAssertions.Json.Specs/JsonAssertionExtensionsSpecs.cs [hunk 1]
```diff
@@ -1,10 +1,8 @@
 using Newtonsoft.Json.Linq;
-using FluentAssertions;
-using FluentAssertions.Json;
 using Xunit;
 
 // NOTE that we are using both namespaces 'FluentAssertions' & 'FluentAssertions.Json' from an external namespace to force compiler disambiguation warnings
```
[the remark this answers is not in the document]
Comment (dennisdoomen) [reply Tests/FluentAssertions.Json.Specs/JsonAssertionExtensionsSpecs.cs:4]: I restored what was there before. No time to look at right now.

**Comment by dennisdoomen** [comment]
> By comparing the nupkg from this CI build with the newest from nuget.org, it seems something with assembly versioning is not working correctly.

You must be looking at the wrong place, because this is what I see when I download the package from the CI build:

image

**Comment by jnyrup** [comment]
Try having a look at the dll in the net451 folder

**Comment by jnyrup** [comment]
The problems seems to be `<GenerateAssemblyInfo>false</GenerateAssemblyInfo>` in the csproj.
This commit should fix it.
https://github.com/jnyrup/fluentassertions.json/commit/771da2806a437411d30f6dd2e841151ffc8def4f

That commit will also fix, that the nuspec and csproj where inconsistent in package versions.

**Comment by dennisdoomen** [comment]
> Try having a look at the dll in the net451 folder

I can always trust your sharp eye.

Didn't even see your follow-up message.

**Comment by jnyrup** [comment]
It seems your latest changes fixed the Assembly Version problem.
Do you intend to use my suggestion commit, or should I do that in a follow up PR?

**Comment by dennisdoomen** [comment]
> Do you intend to use my suggestion commit, or should I do that in a follow up PR?

Please provide a follow-up commit so the other folks can continue.

**Pull request merged**